data-prepper-plugins/s3-source/README.md

```diff
@@ -72,13 +78,27 @@ All Duration values are a string that represents a duration. They support ISO_86
 
 * `expression` (Required if s3_select enabled) : Provide s3 select query to process the data using S3 select for the particular bucket.
 
+* `expression_type` (Optional if s3_select enabled) : Provide s3 select query type to process the data using S3 select for the particular bucket.
+
+* `compression_type` (Optional if s3_select enabled) : The compression algorithm to apply. May be one of: `none`, `gzip`. Defaults to `none`.
+
 * `input_serialization` (Required if s3_select enabled) : Provide the s3 select file format (csv/json/Apache Parquet) Amazon S3 uses this format to parse object data into records and returns only records that match the specified SQL expression. You must also specify the data serialization format for the response.
 
-* `csv_file_header` (Optional) : Provide CSV Header example : `use` , `none` , `ignore`. Default is `use`.
+* `csv` (Optional) : Provide the csv configuration to process the csv data.
+
+* `file_header_info` (Required if csv block is enabled) : Provide CSV Header example : `use` , `none` , `ignore`. Default is `use`.
+
+* `quote_escape` (Optional) : Provide quote_escape attribute example : `,` , `.`.
+
+* `comments` (Optional) : Provide comments attribute example : `#`. Default is `#`.
+
+* `json` (Optional) : Provide the json configuration to process the json data.
+
+* `type` (Optional) : Provide the type attribute to process the json type data example: `Lines` , `Document` Default is `Document`.
 
 * `notification_type` : Must be `sqs`.
 
-* `compression` (Optional) : The compression algorithm to apply. May be one of: `none`, `gzip`, or `automatic`. Defaults to `none`.
+* `compression_type` (Optional if sqs enabled) : The compression algorithm to apply. May be one of: `none`, `gzip`, or `automatic`. Defaults to `none`.
```

Comment:
@udaych20 , Please also correct this back to the original value.

Also, this should just be `(Optional)`. It shouldn't matter if SQS is enabled or not.